src/gui/explorer.go

```diff
@@ -155,21 +152,66 @@ func getTransactionsForAddress(gateway *daemon.Gateway) http.HandlerFunc {
 			return
 		}
 
+		// Get start and end pagination index
+		start := r.FormValue("start")
+		startIndex, err := strconv.Atoi(start)
+		if (start != "" && err != nil) || startIndex < 0 {
+			if startIndex < 0 {
+				err = errors.New("start index cannot be negative")
+			}
+			wh.Error400(w, fmt.Sprintf("invalid start index, %s: %v", start, err))
+			return
+		}
+
+		end := r.FormValue("end")
+		endIndex, err := strconv.Atoi(end)
+		if (end != "" && err != nil) || endIndex < 0 {
+			if endIndex < 0 {
+				err = errors.New("end index cannot be negative")
+			}
+			wh.Error400(w, fmt.Sprintf("invalid end index, %s: %v", end, err))
+			return
+		}
+
 		cipherAddr, err := cipher.DecodeBase58Address(addr)
 		if err != nil {
 			wh.Error400(w, "invalid address")
 			return
 		}
 
 		txns, err := gateway.GetAddressTxns(cipherAddr)
+		totalLen := len(txns.Txns)
```

Comment:
This reduces the amount of data sent in the request but the real bottleneck is the db query, is it possible to improve query performance with the range parameters?